client/src/pages/CodeUploadDialog/CodeUploadDialog.jsx

```diff
@@ -32,6 +37,9 @@ const CodeUploadDialog = ({ open, onClose }) => {
   const [ language, setLanguage ] = useState(defaultLanguage);
   const [ codeSnippet, setCodeSnippet ] = useState('');
   const [ comments, setComments ] = useState('');
+  const [ errorSnackbar, setErrorSnackbar ] = useState({ open: false, message: '' });
+  const [ successSnackbar, setSuccessSnackbar ] = useState({ open: false, message: '' });
```

Comment:
You could probably consolidate both of these snackbars into one snackbar by adding a field called severity (and that could be either success or error

This would reduce a lot of code and reduce the need for many of the functions below too